bazel/grpc_extra_deps.bzl

```diff
@@ -53,7 +56,7 @@ def grpc_extra_deps(ignore_version_differences = False):
     api_dependencies()
 
     go_rules_dependencies()
-    go_register_toolchains(version = "1.18")
+    go_register_toolchains(version = go_version if go_version != "host" else None)
```

Comment:
So if `go_version` is `"host"` here, we pass `None` to `go_register_toolchains`? This seems like it would be used for [registering a custom toolchain](https://github.com/bazelbuild/rules_go/blob/master/go/toolchains.rst#registering-a-custom-sdk), not using [the installed toolchain](https://github.com/bazelbuild/rules_go/blob/master/go/toolchains.rst#using-the-installed-go-sdk).

Is there any reason not to just pass the `go_version` kwarg of `grpc_extra_deps` unmodified to `go_register_toolchains`?

Reply:
> So if go_version is "host" here, we pass None to go_register_toolchains? 

Yes

> This seems like it would be used for [registering a custom toolchain](https://github.com/bazelbuild/rules_go/blob/master/go/toolchains.rst#registering-a-custom-sdk), not using [the installed toolchain](https://github.com/bazelbuild/rules_go/blob/master/go/toolchains.rst#using-the-installed-go-sdk).

I was not aware of this but it makes using "host" confusing. Alternatively I could use another string such as `bazel-toolchains`.

> Is there any reason not to just pass the go_version kwarg of grpc_extra_deps unmodified to go_register_toolchains?

It would not work as by default `grpc_extra_deps` passes `version = "1.18"` to `go_register_toolchains`, making `go_version` useless. 

From [rules_go source code](https://github.com/bazelbuild/rules_go/blob/8d309d5af15814b4096d80b60f80fa86128c43f2/go/private/sdk.bzl#L544):
```
def go_register_toolchains(version = None, nogo = None, go_version = None, experiments = None):
    """See /go/toolchains.rst#go-register-toolchains for full documentation."""
    if not version:
        version = go_version  # old name
```

Reply:
bump